### PR TITLE
Allow changing spreedly base url in config

### PR DESCRIPTION
**Pull request by lasergoat**
Addresses Issue #30 

In order to support a proxy service or other scenarios, this PR allows the base url to not be spreedly. 

This is a simple change on the Client.php class. 

Example use:

```
$spreedlyConfig = [
    "baseUrl" => "http://someproxy.com/",
    "key" => "aaaaa",
    "secret" => "bbbbb"
];

Spreedly::setConfig($spreedlyConfig); // - OR - 
Spreedly::mergeConfig($spreedlyConfig);
```

The default value is still the real Spreedly api url. 

-----

This PR also enhances the readme with some development and testing instructions.

It also adds a very simple `package.json` which helps with install and running tests.

```npm install
npm test // simply runs phpspec
```

### CODE REVIEW

**Comment by lasergoat** [comment]
It looks like my prettier formatter also adjusted the formatting on the Readme, let me know if you'd like me to undo those formatting changes.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "test": "vendor/bin/phpspec run",
+    "install": "composer install"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/fattmerchantorg/spreedly.git"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
Comment (tuurbo) [inline package.json]: should be MIT

File: package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "spreedly",
+  "version": "1.0.0",
+  "description": "**[changelog](#changelog) !! As of the 2.0 release the amount must be an integer as required by Spreedly. E.g., 1098 for $10.98 !!**",
+  "main": "index.js",
```
Comment (tuurbo) [inline package.json]: please remove "main"

File: package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "spreedly",
+  "version": "1.0.0",
```
Comment (tuurbo) [inline package.json]: remove version?

**Comment by tuurbo** [comment]
> It looks like my prettier formatter also adjusted the formatting on the Readme, let me know if you'd like me to undo those formatting changes.

no need

File: src/Client.php [hunk 4]
```diff
@@ -26,6 +26,12 @@ class Client
      */
     public function __construct(GuzzleInterface $client, $config)
     {
+        // allow overriding BASE_URL
+        if (!isset($config['baseUrl']))
+        {
```
Comment (tuurbo) [inline src/Client.php]: bracket should be on same line as "if"

**Comment by lasergoat** [comment]
@tuurbo good catches, I have fixed the issues you mentioned.

**Comment by tuurbo** [comment]
@lasergoat It is now merged, Thanks :)

**Pull request merged**